### PR TITLE
[Chore] - Migrated MiddlewareListener to TS

### LINKED ISSUE #1538: MiddlewareListener.js
file: https://github.com/RedHatInsights/frontend-components/tree/master/packages/utils/src/MiddlewareListener/MiddlewareListener.js

### PR DESCRIPTION
**Pull request by adithyaakrishna**
Signed-off-by: Adithya Krishna <adikrish@redhat.com>

Description:
This PR fixes #1538

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@adithyaakrishna  You have to define the Listener type as well. Fortunately, the shape is strictly defined. You can follow the diff:

```diff
diff --git a/packages/utils/src/MiddlewareListener/MiddlewareListener.ts b/packages/utils/src/MiddlewareListener/MiddlewareListener.ts
index 191499ef..ade526ca 100644
--- a/packages/utils/src/MiddlewareListener/MiddlewareListener.ts
+++ b/packages/utils/src/MiddlewareListener/MiddlewareListener.ts
@@ -1,10 +1,14 @@
 import { Dispatch, AnyAction } from 'redux';
 
-/* eslint-disable @typescript-eslint/no-explicit-any */
+export type Listener = {
+  callback: (...args: unknown[]) => void;
+  on: string;
+};
+
 export class MiddlewareListener {
-  [x: string]: any;
+  listeners: Set<Listener>;
   constructor() {
-    this.listeners = new Set();
+    this.listeners = new Set<Listener>();
   }
 
   getListeners() {
@@ -12,7 +16,7 @@ export class MiddlewareListener {
   }
 
   getMiddleware() {
-    return () => (next: Dispatch<AnyAction>) => (action: { type: any; payload: any }) => {
+    return () => (next: Dispatch<AnyAction>) => (action: AnyAction) => {
       const preventBubble = this.callOnAction(action.type, action.payload);
       if (preventBubble) {
         next({ type: '@@config/action-stopped', payload: action });
@@ -22,12 +26,12 @@ export class MiddlewareListener {
     };
   }
 
-  addNew(listener: Record<string, never>) {
+  addNew(listener: Listener) {
     this.listeners.add(listener);
     return () => this.listeners.delete(listener);
   }
 
-  callOnAction(action: any, data: any) {
+  callOnAction(action: string, data: unknown) {
     let stopBubble = false;
     const preventBubble = () => (stopBubble = true);
     const listeners = Array.from(this.listeners);

```

**Comment by adithyaakrishna** [comment]
@Hyperkid123 Ack, Thank you, @Hyperkid123 🚀

File: packages/utils/src/MiddlewareListener/MiddlewareListener.ts [hunk 1]
```diff
@@ -1,14 +1,23 @@
+import { Dispatch, AnyAction } from 'redux';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (Hyperkid123) [inline packages/utils/src/MiddlewareListener/MiddlewareListener.ts:3]: Can you remove this line?

**Comment by Hyperkid123** [comment]
Also the test files have some TS errors

**Comment by adithyaakrishna** [comment]
@Hyperkid123 Tests successful and all green ✅

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**